tests/testthat/test-markdown-code.R

```diff
@@ -1,33 +1,50 @@
-
-test_that("can eval", {
-  out1 <- roc_proc_text(rd_roclet(), "
-    #' @title Title `r 1 + 1`
-    #' @description Description `r 2 + 2`
-    #' @md
-    foo <- function() {}")[[1]]
-
+test_that("can eval inline code", {
+  example <- test_path("roxygen-block-md-code-simple-inline.R")
+  out1 <- roc_proc_text(rd_roclet(), brio::read_file(example))[[1]]
   expect_equal(out1$get_value("title"), "Title 2")
   expect_equal(out1$get_value("description"), "Description 4")
 })
 
-test_that("uses the same env for a block, but not across blocks", {
+test_that("can eval fenced code", {
+  example <- test_path("roxygen-block-md-code-simple-fenced.R")
+  out1 <- roc_proc_text(rd_roclet(), brio::read_file(example))[[1]]
+  expect_match(out1$get_value("details"), "2")
+})
+
+test_that("use same env within, but not across blocks", {
+  example <- test_path("roxygen-block-md-code-envs.R")
+  out1 <- roc_proc_text(rd_roclet(), brio::read_file(example))[[1]]
+  out2 <- roc_proc_text(rd_roclet(), brio::read_file(example))[[2]]
+  expect_equal(out1$get_value("title"), "Title  420")
+  expect_equal(out1$get_value("description"), "Description TRUE")
+  expect_equal(out2$get_value("description"), "Description FALSE")
+})
+
+test_that("appropriate knit print method for fenced and inline is applied", {
+  knit_print.foo <- function(x, inline = FALSE, ...) {
+    knitr::asis_output(ifelse(inline, "inline", "fenced"))
+  }
+  withr::local_package("knitr")
+  registerS3method(
+    genname = "knit_print", 
+    class = "foo", 
+    method = "knit_print.foo"
+  )
```

Comment:
I think we usually write this as:

```R
local_bindings(
  knit_print.foo = function(x, inline = FALSE, ...) {
     knitr::asis_output(ifelse(inline, "inline", "fenced"))
   },
  .env = globalenv()
)
```

(Unfortunately `local_bindings()` currently comes from rlang, not withr)

This avoids the need to register by putting the method in the global env, and ensures that it's automatically cleaned up afterwards. (It's unlikely that the dangling S3 method registration will cause problems, but stranger things have happened.)